docs/operator_tutorial.md

```diff
@@ -0,0 +1,209 @@
+
+```javascript
+{
+     profileInfo: {
+       profileID: 'pr-12333211',
+       userName: 'JohnyAppleseed',
+     },
+     address: {
+       line1: '123 Easy St.',
+       line2: '',
+       city: 'Athens',
+       stateProvince: 'GA',
+       postalCode: '30606',
+       country: 'USA',
+     },
+     social: {},
+     attributes: {},
+   }
+```
+
+A subtle problem is the [setUserVars function](https://developer.fullstory.com/user-variables) expects a simple object and not one that contains several depths of information. The properties `profileInfo`, `address`, `social`, and `attributes` are examples of properties that contain additional properties at a different depth.  The `FS.setUserVars` argument should be an object with no children and only name value pairs.
+
+To fix this problem, the **flatten** operator is used to copy all properties at any depth into a single object.  This resolves the incompatibility, and the rule appears as the following and produces the next result.
+
+```javascript
+   {
+     id: "fs-uservars-ceddl-user-all",
+     description: "send all CEDDL user properties to FS.setUserVars",
+     source: "digitalData.user.profile[0]",
+     operators: [ { name: "flatten" } ],
+     destination: "FS.setUserVars"
+   }
+```
+
+```javascript
+{
+ profileID: 'pr-12333211',
+ userName: 'JohnyAppleseed',
+ line1: '123 Easy St.',
+ line2: '',
+ city: 'Athens',
+ stateProvince: 'GA',
+ postalCode: '30606',
+ country: 'USA'
+ }
+```
+
+Note that all properties now exist at the same level and properties with empty data have been removed.
+
+## Creating Function Arguments with Insert
+
+Now consider this, rather than calling `FS.setUserVars`, you want to call `FS.identify`.  The [identify function](https://developer.fullstory.com/identify) requires not one but two arguments: `identify(uid, object)` where `uid` is a string and the `object` is similar to the one used with `setUserVars`.  To correctly call the function, you need a string value for the first argument and an object for the second.  This use case can be accomplished with the following rule.
+
+```javascript
+{
+  id: "fs-identify-ceddl-user-all",
+  description: "send all CEDDL user properties to FS.identify using the profileID as the FullStory uid",
+  source: "digitalData.user.profile[0]",
+  operators: [ { name: "flatten" } , { name: "insert", select: "profileID" } ],
+  destination: "FS.identify"
+}
+```
+
+The first profile element still needs to be flattened, and this operator remains first.  An **insert** operator is added next.  Each time you add an operator, data is transformed by the previous operator and is passed to the next operator in the sequence.  **Insert** adds a value to create a list of arguments (parameters) that are sent to functions.  The insert operator is configured using a `select` option set to `profileID`.  The `select` option uses selector syntax to find the value at `profileID` in the flattened object and insert it.  The result is a list.
```

Comment:
Just a suggestion, but it might be helpful if you broke down the intermediate state of the output object. Something like "After `flatten` the output looks like this... Then after `insert` it looks like this... The output state after the final operator is what is passed into the destination"

Or something like that.

Reply:
Opened as issue #60